code/game/atoms_movable.dm

```diff
@@ -469,6 +469,67 @@
 	return
 
 
+/atom/movable/proc/do_attack_animation(atom/A, visual_effect_icon, obj/item/used_item, no_effect)
```

Comment:
We already had `Mob.animation_attack_on(atom/A)`
Wont this conflict?

Reply:
It won't conflict, our code should be refactored to use this in the long term though.